includes/class-internal-events.php

```diff
@@ -255,6 +256,34 @@ private function migrate_legacy_cron_events() {
 		}
 	}
 
+	// Recurring events that have the same action/args/schedule are unnecessary. We can safely remove them.
+	private function prune_duplicate_events() {
+		$events = Events::query( [ 'limit' => -1, 'orderby' => 'ID', 'order' => 'ASC' ] );
+
+		$original_events  = [];
+		$duplicate_events = [];
+		foreach ( $events as $event ) {
+			if ( ! $event->is_recurring() ) {
+				// Only interested in recurring events.
```

Comment:
Is this because we may have single scheduled events with the same args that are supposed to run synchronously?

Reply:
Sorta. Wrote some about it in the description:

> I opted for not including duplicate single events, since they technically clean themselves up. But could possibly revisit that in the future. For those will probably want to respect the "10minute window" that core checks: https://developer.wordpress.org/reference/functions/wp_schedule_single_event/. As in, only remove duplicate single events that exist within 10minutes of each other.